kernel/src/memop.rs

```diff
@@ -52,48 +52,48 @@ pub(crate) fn memop(process: &dyn Process, op_type: usize, r1: usize) -> Syscall
         // Op Type 1: SBRK
         1 => process
             .sbrk(r1 as isize)
-            .map(|addr| SyscallReturn::SuccessU32(addr as u32))
+            .map(|addr| SyscallReturn::SuccessPtr(addr))
```

Comment:
Except this one, which is a `Ptr` rather than a `usize` and is a new return variant.

Reply:
...although `into_compat` will reverse this again.